core/os/input_event.cpp

```diff
@@ -34,8 +34,52 @@
  */
 
 bool InputEvent::operator==(const InputEvent &p_event) const {
+	if (type != p_event.type){
+		return false;
+	}
 
-	return true;
+	switch(type) {
```

Comment:
I think it's okay to keep it in one commit, but maybe you can amend the log to mention the two changes? `git commit --amend`.